modules/core/src/smithy4s/HintMask.scala

```diff
@@ -0,0 +1,36 @@
+package smithy4s
+
+abstract class HintMask {
```

Comment:
Could this just be a newtype over Hints?

Reply:
nope, Hints contain keys and values, HintMask just contains keys

Reply:
Ah true 😂